# A distinct epigenetic profile distinguishes stenotic from non-inflamed fibroblasts in the ileal mucosa of Crohnâ€™s disease patients


library(SummarizedExperiment)
library(ggplot2)
library("pheatmap")
library("RColorBrewer")
library(clusterProfiler)
library(org.Hs.eg.db)


if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("DESeq2")


library(DESeq2)

#Load the gene expression matrix
df <- read.delim("GENDX000019_GeneMatrix_rawCounts.txt", row.names=1)

#Load the metadata and delete the non useful columns
metadata <- read.delim("GEND000019-meta.txt", row.names=5)
# met<- as.data.frame(lapply(met, as.factor))


# Remove features that do not have sufficient reads
df1 <- df[rowSums(df>1) >= 11,]
genes <- row.names(df1)

df1<- as.data.frame(lapply(df1, as.integer))

#Set as rownames only the Ensembl IDs (cutting the part of gene names)
row.names(df1) <- substr(genes, 1, 15)

#metadata$Age <- as.factor(metadata$Age)
metadata$Age.Level <-  cut(metadata$Age, c(17, 40, 80))


#DESeq

#INTRA
dds1 <- DESeqDataSetFromMatrix(df1, colData=metadata, design = ~ Gender + Age.Level + Development.Stage + Disease)

# log
rld1 <- rlog(dds1, blind=FALSE)

plotPCA(rld1, intgroup = c("Age.Level"))

dds1 <- DESeq(dds1)

res1<-results(dds1)

summary(res1)

# Show the significant genes with the strongest down or up regulation
resSig1 <- subset(res1, padj < 0.1)
resSig1 <- resSig1[order(resSig1$log2FoldChange, decreasing=T), ]
head(resSig1)

# Show significant effects of treatment on gene counts more than doubling or less than halfing
resLFC1 <- results(dds1, lfcThreshold=1)
table(resLFC1$padj < 0.1)

#Enrichment (da sistemare)
gene.sym <- bitr(row.names(res1), fromType = 'ENSEMBL', toType = 'SYMBOL', OrgDb = org.Hs.eg.db)

#geneList_DESeq1 <- resSig1$log2FoldChange
#names(geneList_DESeq1) <- row.names(resSig1)
#kk1 <- gseKEGG(geneList = geneList_DESeq1, organism = 'hsa', keyType = "ENSEMBL", pvalueCutoff = 0.1, verbose = FALSE)
#head(kk1)





#INTER
dds2 <- DESeqDataSetFromMatrix(df1, colData=metadata, design = ~ Gender + Age.Level + Development.Stage + Disease.State)

# log
rld2 <- rlog(dds2, blind=FALSE)

plotPCA(rld2, intgroup = c("Disease.State"))

dds2 <- DESeq(dds2)

res2<-results(dds2)

summary(res2)

# Show the significant genes with the strongest down or up regulation
resSig2 <- subset(res2, padj < 0.1)
head(resSig2[order(resSig2$log2FoldChange, decreasing=T), ])

# Show significant effects of treatment on gene counts more than doubling or less than halfing
resLFC2 <- results(dds2, lfcThreshold=1)
table(resLFC2$padj < 0.1)

# We can also make custom plots using the ggplot function For example we can show the best gene based on treatment and cell line:
topGene2 <- rownames(res2)[which.min(res2$padj)]

geneCounts2 <- plotCounts(dds2, gene=topGene2, intgroup=c("Disease", "Disease.State"), returnData=TRUE)


ggplot(geneCounts2, aes(x=Disease, y=count, color=Disease.State)) +
  scale_y_log10() + 
  geom_point(position=position_jitter(width=.1,height=0), size=3)



#ALL
dds3 <- DESeqDataSetFromMatrix(df1, colData=metadata, design = ~ Gender + Age.Level + Development.Stage + Disease + Disease.State
        + Disease:Disease.State)
dds3 <- DESeq(dds3)
        

dds4 <- DESeqDataSetFromMatrix(df1, colData=metadata, design = ~ Gender + Age.Level + Development.Stage + Disease)
dds4 <- DESeq(dds4)


dds5 <- DESeqDataSetFromMatrix(df1, colData=metadata, design = ~ Gender + Age.Level + Development.Stage + Disease)
dds5 <- DESeq(dds5)


